### PR TITLE
Some introduction PDF files of Motr and HLD of SNS Repair

### PR DESCRIPTION
**Pull request by TechWriter-Mayur**
* HLD of SNS Repair.
* Introduction of Motr on SAGE platform.
* Technical Presentation of Motr.

Signed-off-by: TechWriter-Mayur <mayur.gupta@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by huanghua78** [comment]
Please fix the signature.

**Comment by huanghua78** [comment]
I'd like @sainarasim to review and confirm this PR, whether these documents are OK to be uploaded to our repo.

**Review by huanghua78 — APPROVED** [review]
Looks fine.

Also, I got approval from Sai.

**Comment by huanghua78** [comment]
Hello @TechWriter-Mayur, please contact @nikitadanilov, to find the source file of doc/PDF/motr_technical_long.pdf,
and update "clovis" to "Motr client", etc.

Clovis is obsolete. We will not use it any longer.

**Comment by TechWriter-Mayur** [comment]
> Hello @TechWriter-Mayur, please contact @nikitadanilov, to find the source file of doc/PDF/motr_technical_long.pdf,
> and update "clovis" to "Motr client", etc.
> 
> Clovis is obsolete. We will not use it any longer.

@huanghua78, I have updated the  motr_technical_long.pdf document.

File: doc/HLD_of_SNS_Repair.rst [hunk 1]
```diff
@@ -0,0 +1,1647 @@
...
+-  Fractions of respectively storage, network and processor bandwidth dedicated to repair are A\ :sub:`S`, A\ :sub:`N` and A\ :sub:`P`
+
+Let's consider a steady state of repair in a pool with F\ :sub:`S` failed servers and F\ :sub:`D` failed devices (F\ :sub:`D` includes all
+devices on failed servers), assuming at most one unit is lost in any parity group. Define G\ :sub:`D`, G\ :sub:`S`, G\ :sub:`P` and G\ :sub:`O` as rates (bytes/sec) at which repair reads data from a device, sends data from a given server, computes redundancy codes and writes data to a device respectively. 
+
+Every one of N\ :sub:`S` - F\ :sub:`S` survived servers has on average (N\ :sub:`D` - F\ :sub:`D`)/(N\ :sub:`S` - F\ :sub:`S`) devices attached
+and from each of these data are read at the rate G\ :sub:`D`. Assuming that none of this data are for "internal consumption" (that is, assuming
+that no parity group has a spare space unit on a server where it has data or parity units),  servers sends out all these data, giving
+
+.. math:: G_{D}^{} \cdot \frac{N_{D}^{} - F_{D}^{}}{N_{S}^{} - F_{S}^{}} = G_{S}^{}
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:1376]: Does this display properly in your web browser?

It does not show any meaningful math in my chrome.
Can you check?

**Comment by huanghua78** [comment]
I propose to remove all dead link in the document, only keeping the link title there.

File: doc/HLD_of_SNS_Repair.rst (hunk 1 above)
Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: updated

File: doc/HLD_of_SNS_Repair.rst [hunk 2]
```diff
@@ -0,0 +1,1622 @@
+High Level Design of SNS Repair
+===============================
+
+
+This document provides a High-Level Design (HLD) of SNS repair for Motr M0. The main purposes of this document are:
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:5]: Remove the "M0".

File: doc/HLD_of_SNS_Repair.rst [hunk 3]
```diff
@@ -0,0 +1,1622 @@
...
+
+- To be inspected by M0 architects and peer designers to make sure that HLD is aligned with M0 architecture and other designs, and contains no defects
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:7]: Replace the "M0" with "Motr".

File: doc/HLD_of_SNS_Repair.rst [hunk 4]
```diff
@@ -0,0 +1,1622 @@
...
+.. **Note:** This document has been revised to exclude the copy machine related information. The copy machine design has been included in a separate document HLD of copy machine and agents.
+
+Redundant striping is a proven technology to achieve higher throughput and data availability. The Server Network Striping (SNS) applies this technology to network devices and achieves similar goals as a local RAID.
+In the case of storage and/or server failure, the SNS repair reconstructs the lost data from survived data reliably and quickly, without major impact to the production systems. This document presents the HLD of SNS repair.
+
+
+1. Definitions
+--------------
+
+.. Definitions of terms and concepts used by the design go here. The definitions must be as precise as possible. References to the `Motr Glossary<Link-to-be-added>`__ are permitted and encouraged. Agreed upon terminology should be incorporated in the glossary.
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:26]: `Motr Glossary </doc/PDF/Glossary.pdf >`__

Comment (huanghua78) [reply doc/HLD_of_SNS_Repair.rst:26]: Actually, I made these changes already, but I think you over-wrote it in your commits.

File: doc/HLD_of_SNS_Repair.rst [hunk 5]
```diff
@@ -0,0 +1,1622 @@
...
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| U           | 1.0    |           |                                                                                                          |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| S\ :sub:`D` | 2.0e12 | bytes     | 2TB drive                                                                                                |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`S` | 4.0e9  | bytes/sec | IB QDR                                                                                                   |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`P` | 8.0e9  | bytes/sec | `check-sum throughput <check-sum-throughput>`__                                                          |
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:1426]: Please remove the fourth column, "`check-sum throughput <check-sum-throughput>`__"

It is not accessible.

File: doc/HLD_of_SNS_Repair.rst [hunk 6]
```diff
@@ -0,0 +1,1622 @@
...
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`D` | 7.0e7  | bytes/sec | `ST31000640SS <ST31000640SS>`__                                                                          |
```
Comment (huanghua78) [inline doc/HLD_of_SNS_Repair.rst:1428]: The same for "`ST31000640SS <ST31000640SS>`__"

Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: Updated.

File: doc/HLD_of_SNS_Repair.rst (hunk 5 above)
Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: Updated.

File: doc/HLD_of_SNS_Repair.rst (hunk 4 above)
Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: Updated.

File: doc/HLD_of_SNS_Repair.rst (hunk 3 above)
Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: Updated.

File: doc/HLD_of_SNS_Repair.rst (hunk 2 above)
Comment (TechWriter-Mayur) [reply doc/HLD_of_SNS_Repair.rst]: Updated.

**Comment by huanghua78** [comment]
I am asking @Saumya-Sunder @johnbent  to review this, too.

**Review by johnbent — APPROVED** [review]
Thanks @huanghua78 .  Looks good to me.

**Pull request merged**